### PR TITLE
Query ores score

### PR DESCRIPTION
**Pull request by Silvan-WMDE**
This PR is based on top of the `refactor` branch and adds the ability to fetch _number of sitelinks_ and _ores score_ in batches. It is quite large overall, but comprises quite a few separate commits, which show the extended functionality when reviewed step-by-step.

### CODE REVIEW

File: checkDataQuality.py [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+OUTPUT_DELIMITER = ';'
+STATEMENT_COUNT_URL = 'https://www.wikidata.org/w/api.php?format=json&action=query&prop=pageprops|revisions&ppprop=wb-claims&rvprop=ids'
+SITELINK_COUNT_URL = 'https://www.wikidata.org/w/api.php?format=json&action=wbgetentities&props=sitelinks'
+CONSTRAINT_CHECK_URL = 'https://www.wikidata.org/w/api.php?format=json&action=wbcheckconstraints'
+ORES_URL = 'https://ores.wikimedia.org/v3/scores/wikidatawiki?revids='
+
+batchSize = 10
+
+def usage(exitCode = False):
+    print('checkConstraints.py -i <inputfile> | -r <number of items> [-o <outputfile> -b <batch-size>]')
```
Comment (botoxparty) [inline checkDataQuality.py:23]: File name is updated to `checkQualityData.py`

**Review by botoxparty — CHANGES_REQUESTED** [review]
Thanks! This was a fun script to play around in today! Just some small comments but everything looks good.

File: checkDataQuality.py (hunk 1 above)
Comment (Silvan-WMDE) [reply checkDataQuality.py]: fixed in [0df29b6](https://github.com/wmde/wikidata-constraints-violation-checker/pull/29/commits/0df29b6bb156a025a5e5267c923b80f251d8ca07)

File: checkDataQuality.py [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+
+def logException(exception):
+    with open('error.log', 'a') as outputFile:
+        print(exception, file=outputFile)
+
+def logErrorMessage(message):
+    with open('error.log', 'a') as outputFile:
+        print(message, file=outputFile)
+
+def displayProgress(step, overwrite=True):
```
[the remark this answers is not in the document]
Comment (Silvan-WMDE) [reply checkDataQuality.py:143]: previous versions, such as the current `main` branch show some progress indicator on the console. For batch processing it hasn't been implemented, though. Feel free to use this function and re-enable the feature.

**Review by botoxparty — APPROVED** [review]

**Pull request merged**